web/static/vendor/components/notification/notification_table.tsx

```diff
@@ -0,0 +1,169 @@
+"use strict";
+import React, {useState, useEffect} from "react";
+import {createColumnHelper, flexRender, getCoreRowModel, useReactTable} from "@tanstack/react-table";
+import type {Notification} from "./custom_types";
+
+
+type DismissButtonProps = {
+    notification: Notification
+    onClick: (notification: Notification) => void,
+}
+
+/**
+ * DismissButton: React component that displays a button to dismiss a notification
+ * @param notification - Notification to dismiss
+ * @param onClick - Function to call when the button is clicked
+ *
+ * @return - The button to dismiss the notification
+ */
+const DismissButton = ({notification, onClick}: DismissButtonProps) => {
+    return <span className={"btn btn-link p-0 m-0"} onClick={() => onClick(notification)}>Dismiss</span>;
+};
+
+const columnHelper = createColumnHelper<Notification>();
+const dateFormatter = new Intl.DateTimeFormat(
+    Intl.DateTimeFormat().resolvedOptions().locale,
+    {
+        year: "numeric",
+        month: "short",
+        day: "numeric",
+    }
+);
+
+/**
+ * NotificationsTableProps: Props for the NotificationsTable component
+ *
+ * @param data - List of notifications to display
+ * @param showRecipient - If true, the table will display the recipient column
+ * @param showDismiss - If true, the table will display the dismiss column
+ * @param onDismiss - Function to call when a notification is dismissed
+ *
+ */
+type NotificationsTableProps = {
+    data: Notification[]
+    showRecipient: boolean,
+    showDismiss: boolean,
+    onDismiss: (notification: Notification) => void,
+}
+
+/**
+ * NotificationsTable: React component to display a table of notifications
+ * @param {NotificationsTableProps} props
+ *
+ * @return - The table of notifications
+ */
+export const NotificationsTable = (props: NotificationsTableProps) => {
+    const [notifications, setNotifications] = useState(props.data);
+
+    // Update the notifications when the props change
+    useEffect(() => {
+        setNotifications(props.data);
+    }, [props.data]);
+
+    const columns = [
+        // The recipient columns displays the full name of the recipient
+        columnHelper.accessor("recipient.name", {
+            header: "Recipient",
+            id: "fullName",
+            enableHiding: true,
+        }),
+        // The objectName column displays the name of the object that triggered the notification
+        columnHelper.accessor("objectName", {
+            header: () => {
+                const title = notifications[0].objectType;
+                return title.charAt(0).toUpperCase() + title.substring(1);
+            },
+            cell: cell => {
+                const text = cell.getValue();
+                const displayedText = text.length < 70 ? text : text.substring(0, 70) + "...";
```

Comment:
would it make since to truncate it by css instead and hover to see tooltip? or if it is too much work not to truncate it and display it in full. My thought is that description message can't be found in the Entity details page so if it has important info to the user it should be there.

Reply:
I cannot truncate the text via css without fixing the width for the columns. The issue is that I do not understand how to do that in the current react-table version (it was much easier in the version used for the datacatalog).

I can however add the hover box relatively easily I think.

Reply:
it works, thank you!